x/programs/examples/imports/pstate/pstate.go

```diff
@@ -260,3 +271,25 @@ func (i *Import) deleteFn(caller *program.Caller, memOffset int32, size int32) (
 
 	return types.ValI32(int32(ptr)), nil
 }
+
+func (i *Import) logFn(caller *program.Caller, memOffset int32, size int32) (*types.Val, error) {
+	memory, err := caller.Memory()
+	if err != nil {
+		i.log.Error("failed to get memory from caller",
+			zap.Error(err),
+		)
+		return nil, err
+	}
+
+	bytes, err := memory.Range(uint32(memOffset), uint32(size))
+	if err != nil {
+		i.log.Error("failed to read args from program memory",
+			zap.Error(err),
+		)
+		return nil, err
+	}
```

Comment:
If this fails, we should probably just log a warning? I'm not entirely sure. I guess it's fine to leave for now until this becomes a problem